### PR TITLE
add turnstile and mailgun to signup (LAND-12)

### PR DESCRIPTION
**Pull request by lucassshanks**
## Motivation
Adds emailworker and turnstile component to landing page. Limited testing as all the testing library documentation is all outdated and unbelievably poor to work with

Got away without using jsdom for running tests but on the list to figure that out next time.

## Tickets

LAND-12

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,8 +25,10 @@
     "prettier-plugin-svelte": "^3.3.2",
     "svelte": "^5.0.0",
     "svelte-check": "^4.0.0",
+    "svelte-turnstile": "^0.9.0",
     "typescript": "^5.7.2",
     "typescript-eslint": "^8.17.0",
-    "vite": "^5.0.3"
+    "vite": "^5.0.3",
+    "lucide-svelte": "^0.468.0"
```
Comment (lucassshanks) [inline package.json]: this is tree-shakeable and efficient to just use an icon from where only that icon is imported. I'm just using their loading indicator and got lazy but if needed i can make my own svg if we think it matters

File: email-worker/vitest.config.mts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { defineWorkersConfig } from '@cloudflare/vitest-pool-workers/config';
```
Comment (lucassshanks) [inline email-worker/vitest.config.mts]: autogenerated. can be used if we want to write tests for the email-worker in the future

File: email-worker/wrangler.toml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+[vars.secrets]
+API_KEY=''
+TURNSTILE_KEY=''
+
+[env.production]
+name = "email-worker-production"
+vars.ENVIRONMENT = "production"
+vars.ALLOWED_ORIGIN_URLS=''
+
+# Bind the Workers AI model catalog. Run machine learning models, powered by serverless GPUs, on Cloudflare’s global network
```
Comment (lucassshanks) [inline email-worker/wrangler.toml]: could delete all this if you want idk

File: email-worker/.prettierrc [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (TroyKomodo) [inline email-worker/.prettierrc]: Do we need a 2nd prettierrc?

File: email-worker/pnpm-lock.yaml [hunk 5]
```diff
@@ -0,0 +1,1764 @@
+lockfileVersion: '9.0'
```
Comment (TroyKomodo) [inline email-worker/pnpm-lock.yaml]: I dont think this is correct, we should be using pnpm workspaces to shasre deps between the website and the worker.It shouldnt generate a 2nd lock file i think.

File: email-worker/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+ * - Open a browser tab at http://localhost:8787/ to see your worker in action
+ * - Run `npm run deploy` to publish your worker
+ *
+ * Bind resources to your worker in `wrangler.toml`. After adding bindings, a type definition for the
+ * `Env` object can be regenerated with `npm run cf-typegen`.
+ *
+ * Learn more at https://developers.cloudflare.com/workers/
+ */
+
+import { validate } from 'email-validator';
```
Comment (TroyKomodo) [inline email-worker/src/index.ts:14]: Does CF workers not have a std library with a utility like this? its fine if not.

File: email-worker/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+			console.error('failed to validate captcha', e);
+			return new Response('Internal Server Error', { status: 500, headers });
+		}
+
+		if (!turnstileOutcome.success) {
+			console.error('failed to validate captcha', turnstileOutcome);
+			return new Response('The provided Turnstile token was not valid!', { status: 401, headers });
+		}
+
+		let result: Response;
```
Comment (TroyKomodo) [inline email-worker/src/index.ts]: I feel like for this last one, we dont have to put it outside the try block.

File: email-worker/test/index.spec.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
+// // test/index.spec.ts
```
Comment (TroyKomodo) [inline email-worker/test/index.spec.ts]: is this needed?

File: email-worker/test/tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (TroyKomodo) [inline email-worker/test/tsconfig.json]: Can we share a tsconfig or not?

File: email-worker/wrangler.toml (hunk 3 above)
Comment (TroyKomodo) [reply email-worker/wrangler.toml]: yea

File: email-worker/test/index.spec.ts (hunk 8 above)
Comment (lucassshanks) [reply email-worker/test/index.spec.ts]: no

File: email-worker/src/index.ts (hunk 7 above)
Comment (lucassshanks) [reply email-worker/src/index.ts]: ye

File: email-worker/src/index.ts (hunk 6 above)
Comment (lucassshanks) [reply email-worker/src/index.ts:14]: idk but i tihknk its' fine

File: email-worker/.prettierrc (hunk 4 above)
Comment (lucassshanks) [reply email-worker/.prettierrc]: no

File: .prettierignore [hunk 10]
```diff
@@ -20,3 +20,6 @@ vite.config.js.timestamp-*
 vite.config.ts.timestamp-*
 
 pnpm-lock.yaml
+
+# Email Worker
+email-worker/**
```
Comment (TroyKomodo) [inline .prettierignore]: why is this ignored on prettier?

**Review by TroyKomodo — APPROVED** [review]

**Pull request merged**